### PR TITLE
Feature: Added more documentation and clean code.

### PR DESCRIPTION
**Pull request by werniq**
In this Pull Request, I have added a lot of documentation to functions and structs in the project, so that users can more easily understand what is going on. 

Additionally, I made these additional changes:
- Renamed variables and funcs
- Reformatted Dockerfile
- Added more descriptive logging when error occurs + handled more errors.
- Implemented commands in Makefile to build/push docker image.
- Added taskfile (which In my opinion is more easy to read than Makefile, want the owner to verify it and if needed remove Make)
- Added ability for user to specify some key variables using flags, since some users may want to specify another port/network interface/etc.

I also would like to open Pull Request which integrates K8S to the project, but it contains updates from this PR. To not duplicate changes, I will do it after we resolve this PR.

### CODE REVIEW

File: cmd/collector/main.go [hunk 1]
```diff
@@ -2,21 +2,33 @@ package main
...
+	var (
+		networkInterface string
+	)
+	flag.StringVar(&networkInterface, "i", defaultNetworkInterfaceDescription, "Identification of the interface")
+
+	// question: Why do we need context here? It is not used in collector.Run, except of ctx.Done, but since it is not
+	// context.WithTimeout (as example) it can not be closed in any way.
+	// Same in c.serializer.SerializePackets(ctx, packetsChan), it can not be closed there as well.
+	// Why not just to remove it?
 	ctx := context.Background()
```
Comment (cebilon123) [inline cmd/collector/main.go:27]: Tbh I don't remember why I passed it there, you can remove it. Or, if you want to, you can implement some graceful shutdown on the application close for the sake of art 😄

**Review by cebilon123 — APPROVED** [review]
I love what you did here, for real man, great job. Just give me an answer if you will implement the graceful shutdown or not, and I will merge it 😄

File: cmd/collector/main.go (hunk 1 above)
Comment (werniq) [reply cmd/collector/main.go:27]: Yes, I will. Probably by capturing signals. I guess I will do another PR with graceful shutdown.

Comment (werniq) [reply cmd/collector/main.go:27]: It should be similar to this: 
```go
signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)

go func() {
        if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
            log.Fatalf("Server failed: %s\n", err)
        }
    }()
log.Println("Server started on :8080")

<-quit
log.Println("Shutdown signal received, shutting down gracefully...")

ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
defer cancel()

if err := srv.Shutdown(ctx); err != nil {
        log.Fatalf("Server forced to shutdown: %s\n", err)
}
```

**Pull request merged**